src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -56,19 +66,51 @@ class Lambda {
     dotcomSurrogateKey
   }
 
-  private def sendFastlyPurgeRequestAndAmpPingRequest(contentId: String, purgeType: PurgeType, serviceId: String, surrogateKey: String, fastlyApiKey: String): Boolean = {
-    if (sendFastlyPurgeRequest(contentId, purgeType, serviceId, surrogateKey, fastlyApiKey))
-      sendAmpPingRequest(contentId)
-    else
-      false
+  // Left[Id] for Delete, Right[Id] for Update
+  def decidePurgeType: PartialFunction[Event, Either[Id, Id]] = {
+    case Event(_, EventType.Update, _, _, Some(EventPayload.Content(content))) => Right(content.id)
+    case Event(_, EventType.RetrievableUpdate, _, _, Some(EventPayload.RetrievableContent(content))) => Right(content.id)
+    case Event(_, EventType.Delete, _, _, Some(EventPayload.RetrievableContent(content))) => Left(content.id)
+  }
+
+  def purgeDelete: Id => Future[List[PurgeResult]] = { id =>
+    val fastlyResult: Future[Boolean] = sendFastlyPurgeRequest(id, Hard, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(id), config.fastlyDotcomApiKey)
+    val ampResult: Future[Boolean] = fastlyResult.flatMap(if (_) sendAmpPingRequest(id) else Future.successful(false))
+    Future.sequence(fastlyResult.map(PurgeResult(_, Dotcom)) :: ampResult.map(PurgeResult(_, AMP)) :: Nil)
+  }
+
+  def purgeUpdate: Id => Future[List[PurgeResult]] = { id =>
+    Future.sequence(
+      sendFastlyPurgeRequest(id, Soft, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(id), config.fastlyDotcomApiKey).map(PurgeResult(_, Dotcom)) ::
+        sendFastlyPurgeRequest(id, Soft, config.fastlyApiNextgenServiceId, makeDotcomSurrogateKey(id), config.fastlyDotcomApiKey).map(PurgeResult(_, Nextgen)) ::
```

Comment:
```suggestion
        sendFastlyPurgeRequest(s"${id}.json", Soft, config.fastlyApiNextgenServiceId, makeDotcomSurrogateKey(id), config.fastlyDotcomApiKey).map(PurgeResult(_, Nextgen)) ::
```

Reply:
Thanks for this, looking at how the fastly url is constructed I think it's the `surrogateKey` we need to add the `.json` to rather than the `contentId` (which is just logged out):

https://github.com/guardian/fastly-cache-purger/blob/24220fd47a185b16de548d49505b721f3da11f87/src/main/scala/com/gu/fastly/Lambda.scala#L116

I've just pushed up this change